linkerd/app/core/src/errors/respond.rs

```diff
@@ -149,6 +156,30 @@ impl SyntheticHttpResponse {
             grpc_status: tonic::Code::NotFound,
             close_connection: false,
             message: Cow::Owned(msg.to_string()),
+            location: None,
+        }
+    }
+
+    pub fn redirect(http_status: http::StatusCode, location: &http::Uri) -> Self {
+        Self {
+            http_status,
+            grpc_status: tonic::Code::NotFound,
+            close_connection: false,
+            message: Cow::Borrowed("redirected"),
+            location: Some(
+                http::header::HeaderValue::from_str(&*location.to_string())
+                    .expect("location must be a valid header value"),
```

Comment:
tiny nit: `HeaderValue` has [a `TryFrom<String>` impl](https://docs.rs/http/latest/http/header/struct.HeaderValue.html#impl-TryFrom%3CString%3E), which would take the string by-value and turn it into a `Bytes`, rather than copying from a borrowed string and then dropping the original string:
```suggestion
                http::header::HeaderValue::try_from(location.to_string())
                    .expect("location must be a valid header value"),
```

Reply:
(we would also need to import `TryFrom` if we apply this change)

Reply:
also, a very tiny nit: `HeaderValue` is imported already:
```suggestion
                HeaderValue::from_str(&*location.to_string())
                    .expect("location must be a valid header value"),
```